app/grandchallenge/notifications/tasks.py

```diff
@@ -0,0 +1,40 @@
+from celery import shared_task
+from django.core.paginator import Paginator
+from django.utils.timezone import now
+
+from grandchallenge.notifications.emails import send_unread_notifications_email
+from grandchallenge.notifications.models import Notification
+from grandchallenge.profiles.models import UserProfile
+
+
+@shared_task
+def send_unread_notification_emails():
+    profiles = UserProfile.objects.filter(
+        receive_notification_emails=True
+    ).order_by("user")

```

Comment:
Maybe we can add a filter here for users with unread notifications to save the lookups below.

Reply:
I think that is not possible because `has_unread_notifications` is a property and not stored in the database, right?

Reply:
You should be able to add something like `"user__notification_set__read=False"` to the filter, then you might also need to add a `.distinct()` before the `order_by`.